### PR TITLE
[Fix] Adds notify config default values

### LINKED ISSUE #10633: 🐛 notify config env missing default values
## 🐛 Bug

The notify config env is missing default values.

## 🦋 Expected Behaviour

If there are no values set, only GC Notify should fail, not the in-app notifications.

## 🕵️ Details

https://github.com/GCTC-NTGC/gc-digital-talent/blob/c2de193c35012e15b5606d3eebc0a659aabdec9d/api/config/notify.php#L18-L39

## 📋 Steps to Reproduce

1. Remove all `GCNOTIFY_TEMPLATE_*` keys from `api/.env`
2. `make refresh-api`
3. Trigger a notification
4. Observe both email and in-app notifications not creating an email or in-app notification

## 🙋‍♀️ Proposed Solution

- Add empty string to notify config env `template` default values. For example:  `'test_email' => env('GCNOTIFY_TEMPLATE_TEST_EMAIL', '')`

- add `$e->getMessage()` to:
https://github.com/GCTC-NTGC/gc-digital-talent/blob/88b8489143ab034a39e0aeb18f0ffc1744b343c1/api/app/Observers/PoolObserver.php#L43
https://github.com/GCTC-NTGC/gc-digital-talent/blob/88b8489143ab034a39e0aeb18f0ffc1744b343c1/api/app/Observers/PoolCandidateObserver.php#L49

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the bug was addressed.

- [ ] If there are no values set, only GC Notify emails fail to send
- [ ] add error message to log in `PoolObserver` and `PoolCandidateObserver`

### PR DESCRIPTION
**Pull request by mnigh**
🤖 Resolves #10633.

## 👋 Introduction

This PR fixes config settings that if there are no values set, only GC Notify should fail, not the in-app notifications.

## 🧪 Testing

Assist reviewers with steps they can take to test that the PR does what it says it does.

1. Remove all `GCNOTIFY_TEMPLATE_*` keys from `api/.env`
2. `make refresh-api`
3. Trigger a notification
4. Observe in-app notifications shows up in UI and that the email job is queued in the `jobs` database table

### CODE REVIEW

File: api/app/Observers/PoolObserver.php [hunk 1]
```diff
@@ -40,7 +40,7 @@ public function updated(Pool $pool): void
                         $user->notify($notification);
                     } catch (Throwable $e) {
                         // best-effort: log and continue
-                        Log::error('Failed to send "new job posted" notification to ['.$user->id.']');
+                        Log::error('Failed to send "new job posted" notification to ['.$user->id.']'.$e->getMessage());
```
Comment (petertgiles) [inline api/app/Observers/PoolObserver.php:43]:
```suggestion
                        Log::error('Failed to send "new job posted" notification to ['.$user->id.'] '.$e->getMessage());
```

File: api/app/Observers/PoolCandidateObserver.php [hunk 2]
```diff
@@ -46,7 +46,7 @@ public function updated(PoolCandidate $poolCandidate): void
                     ));
                 } catch (Throwable $e) {
                     // best-effort: log and continue
-                    Log::error('Failed to send "application status changed" notification to ['.$poolCandidate->id.']');
+                    Log::error('Failed to send "application status changed" notification to ['.$poolCandidate->id.']'.$e->getMessage());
```
Comment (petertgiles) [inline api/app/Observers/PoolCandidateObserver.php:49]:
```suggestion
                    Log::error('Failed to send "application status changed" notification to ['.$poolCandidate->id.'] '.$e->getMessage());
```

Comment (esizer) [inline api/app/Observers/PoolCandidateObserver.php:49]: Nit pick. but maybe a space here? :sweat_smile: 
```suggestion
                    Log::error('Failed to send "application status changed" notification to ['.$poolCandidate->id.'] '.$e->getMessage());
```

File: api/app/Observers/PoolObserver.php (hunk 1 above)
Comment (esizer) [inline api/app/Observers/PoolObserver.php:43]: Some thing here.
```suggestion
                        Log::error('Failed to send "new job posted" notification to ['.$user->id.'] '.$e->getMessage());
```

**Review by esizer — APPROVED** [review]
Looks good :+1: , made minor suggestions but nothing that should block it from being merged.

**Review by petertgiles — APPROVED** [review]
Nice. :+1:

**Pull request merged**